Flaky test_regression_payment_complete_after_refund_to_the_initiator

```
_ test_regression_payment_complete_after_refund_to_the_initiator[matrix-False-0-5] _

raiden_network = [<App 0xEA950eB16E9eD7a522C739715Cba5136E861C31f>, <App 0xa06C6602C34Bc173A0070Fa3445CA99d2a64F1e0>, <App 0xa9e398b9B8...960EB99d2a09141e4>, <App 0x11b522Eb9608deE098B5426DD011a625CC469e53>, <App 0x897d9e75B8Eb927F0221CE90CD180BFa41A67303>]
token_addresses = [b'\xb5+\xe6\x06\xfd\xf4\xe2\xcd\xf2\x8b\xf5\xd5\xf8\xd8\xa0\xc0\x9a\xe5\xd7\xd1']
settle_timeout = 75, deposit = 200

    @pytest.mark.parametrize("number_of_nodes", [5])
    @pytest.mark.parametrize("channels_per_node", [0])
    def test_regression_payment_complete_after_refund_to_the_initiator(
        raiden_network, token_addresses, settle_timeout, deposit
    ):
        """Regression test for issue #3915"""
        raise_on_failure(
            raiden_network,
            run_regression_payment_complete_after_refund_to_the_initiator,
            raiden_network=raiden_network,
            token_addresses=token_addresses,
            settle_timeout=settle_timeout,
>           deposit=deposit,
        )

raiden/tests/integration/test_regression.py:262: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
raiden/tests/utils/detect_failure.py:30: in raise_on_failure
    result.get()
src/gevent/event.py:381: in gevent._event.AsyncResult.get
    ???
src/gevent/event.py:409: in gevent._event.AsyncResult.get
    ???
src/gevent/event.py:399: in gevent._event.AsyncResult.get
    ???
src/gevent/event.py:379: in gevent._event.AsyncResult._raise_exception
    ???
../venv-3.7-LINUX/lib/python3.7/site-packages/gevent/_compat.py:47: in reraise
    raise value.with_traceback(tb)
src/gevent/greenlet.py:716: in gevent._greenlet.Greenlet.run
    ???
raiden/tests/integration/test_regression.py:290: in run_regression_payment_complete_after_refund_to_the_initiator
    initiator_app=app0, target_app=app2, token_address=token, amount=deposit, identifier=2
raiden/tests/utils/transfer.py:102: in transfer
    timeout=timeout,
raiden/tests/utils/transfer.py:161: in _transfer_unlocked
    wait_for_unlock.get()
src/gevent/event.py:381: in gevent._event.AsyncResult.get
    ???
src/gevent/event.py:406: in gevent._event.AsyncResult.get
    ???
src/gevent/event.py:117: in gevent._event._AbstractLinkable._wait_core
    ???
src/gevent/event.py:119: in gevent._event._AbstractLinkable._wait_core
    ???
src/gevent/_greenlet_primitives.py:59: in gevent.__greenlet_primitives.SwitchOutGreenletWithLoop.switch
    ???
src/gevent/_greenlet_primitives.py:59: in gevent.__greenlet_primitives.SwitchOutGreenletWithLoop.switch
    ???
src/gevent/_greenlet_primitives.py:63: in gevent.__greenlet_primitives.SwitchOutGreenletWithLoop.switch
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   gevent.timeout.Timeout: 10 seconds

src/gevent/__greenlet_primitives.pxd:35: Timeout
```

Seen in https://circleci.com/gh/raiden-network/raiden/69444#tests/containers/11.

[raiden-debug_2019-08-05T11:42:56.882572.log](https://github.com/raiden-network/raiden/files/3467450/raiden-debug_2019-08-05T11.42.56.882572.log)

--- a/raiden/tests/integration/test_regression.py
+++ b/raiden/tests/integration/test_regression.py
@@ -11,14 +11,18 @@ from raiden.messages.transfers import Lock, LockedTransfer, RevealSecret, Unlock
 from raiden.tests.fixtures.variables import TransportProtocol
 from raiden.tests.integration.fixtures.raiden_network import CHAIN, wait_for_channels
 from raiden.tests.utils.detect_failure import raise_on_failure
-from raiden.tests.utils.events import raiden_state_changes_search_for_item, search_for_item
+from raiden.tests.utils.events import (
+    raiden_events_search_for_item,
+    raiden_state_changes_search_for_item,
+    search_for_item,
+)
 from raiden.tests.utils.factories import UNIT_CHAIN_ID
 from raiden.tests.utils.network import payment_channel_open_and_deposit
 from raiden.tests.utils.transfer import get_channelstate, transfer
 from raiden.transfer import views
-from raiden.transfer.mediated_transfer.events import SendSecretReveal
+from raiden.transfer.mediated_transfer.events import EventRouteFailed, SendSecretReveal
 from raiden.transfer.mediated_transfer.state_change import ReceiveTransferCancelRoute
-from raiden.utils import sha3
+from raiden.utils import PaymentID, sha3
 
 # pylint: disable=too-many-locals
 
@@ -247,7 +251,6 @@ def test_regression_register_secret_once(secret_registry_address, deploy_service
     assert previous_nonce == deploy_service.client._available_nonce
 
 
-@pytest.mark.skip(reason="flaky, see https://github.com/raiden-network/raiden/issues/4532")
 @pytest.mark.parametrize("number_of_nodes", [5])
 @pytest.mark.parametrize("channels_per_node", [0])
 def test_regression_payment_complete_after_refund_to_the_initiator(
@@ -255,8 +258,8 @@ def test_regression_payment_complete_after_refund_to_the_initiator(
 ):
     """Regression test for issue #3915"""
     raise_on_failure(
-        raiden_network,
-        run_regression_payment_complete_after_refund_to_the_initiator,
+        raiden_apps=raiden_network,
+        test_function=run_regression_payment_complete_after_refund_to_the_initiator,
         raiden_network=raiden_network,
         token_addresses=token_addresses,
         settle_timeout=settle_timeout,
@@ -274,23 +277,35 @@ def run_regression_payment_complete_after_refund_to_the_initiator(
     # Topology:
     #
     #  0 -> 1 -> 2
-    #  v         ^
+    #  |         ^
+    #  v         |
     #  3 ------> 4
 
     app_channels = [(app0, app1), (app1, app2), (app0, app3), (app3, app4), (app4, app2)]
-
     open_and_wait_for_channels(app_channels, registry_address, token, deposit, settle_timeout)
 
     # Use all deposit from app1->app2 to force a refund
     transfer(
-        initiator_app=app1, target_app=app2, token_address=token, amount=deposit, identifier=1
+        initiator_app=app1,
+        target_app=app2,
+        token_address=token,
+        amount=deposit,
+        identifier=PaymentID(1),
     )
 
     # Send a transfer that will result in a refund app1->app0
     transfer(
-        initiator_app=app0, target_app=app2, token_address=token, amount=deposit, identifier=2
+        initiator_app=app0,
+        target_app=app2,
+        token_address=token,
+        amount=deposit,
+        identifier=PaymentID(2),
+        timeout=20,
     )
 
     assert raiden_state_changes_search_for_item(
         raiden=app0.raiden, item_type=ReceiveTransferCancelRoute, attributes={}
+    )
+    assert raiden_events_search_for_item(
+        raiden=app0.raiden, item_type=EventRouteFailed, attributes={}
     )

